test/control-component-spec.js

```diff
@@ -8,1512 +8,1591 @@ import { Provider } from 'react-redux';
 import thunk from 'redux-thunk';
```

Comment:
Again a nasty diff to read, sorry!  The most notable changes were: 

1. adding the testContext (of course)  
2. adding the getInitialState() function for setting up the state of the reducers before each set of tests
3. adding the use of get() when getting values out of the model state such as: `get(store.getState().test, 'foo')`